lib/ansible/cli/__init__.py

```diff
@@ -553,39 +552,10 @@ def get_host_list(inventory, subset, pattern='all'):
 
     @staticmethod
     def get_password_from_file(pwd_file):
+        file_vault_secret = get_file_vault_secret(filename=pwd_file)
```

Comment:
The `loader` parameter is required. This wasn't caught by tests because there are none that use this function.

Tests should be added for `--connection-password-file` and `--become-password-file` in a separate PR before continuing with this one. By using a separate PR we can make sure the tests pass with the current code before making changes to it. The case of using `-` as a path should be included in the tests.

Reply:
> The loader parameter is required. This wasn't caught by tests because there are none that use this function.

Oh, funny. I thought that was were the code path went.

I wouldn't have looked here if it weren't for the failed tests :smile: 

https://dev.azure.com/ansible/ansible/_build/results?buildId=46385&view=logs&j=0de013c3-539b-5408-ba9a-bdc827b0cb32&t=4b19883a-28c0-54ce-f1f9-895636b71fa8
```
01:14 # new format, view, using password script
01:14 ansible-vault view "$@" --vault-password-file password-script.py format_1_1_AES256.yml
01:14 + ansible-vault view -vvvvvv --vault-password-file password-script.py format_1_1_AES256.yml
01:14 ERROR! Decryption failed (no vault secrets were found that could decrypt) on format_1_1_AES256.yml for format_1_1_AES256.yml
01:14 ansible-vault [core 2.14.0.dev0]
01:14   config file = /root/ansible/test/results/.tmp/integration/ansible-vault-jbtvhg5w-ÅÑŚÌβŁÈ/test/integration/integration.cfg
01:14   configured module search path = ['/root/ansible/test/support/integration/plugins/modules']
01:14   ansible python module location = /root/ansible/lib/ansible
01:14   ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
01:14   executable location = /root/ansible/bin/ansible-vault
01:14   python version = 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] (/root/ansible/test/results/.tmp/output_dir/venv/bin/python)
01:14   jinja version = 3.0.3
01:14   libyaml = True
01:14 Using /root/ansible/test/results/.tmp/integration/ansible-vault-jbtvhg5w-ÅÑŚÌβŁÈ/test/integration/integration.cfg as config file
01:14 Reading vault password file: /root/ansible/test/results/.tmp/integration/ansible-vault-jbtvhg5w-ÅÑŚÌβŁÈ/test/integration/targets/ansible-vault/unicorn-password-script.py
01:14 Found a vault_id (default) in the vaulttext
01:14 We have a secret associated with vault id (default), will try to use to decrypt format_1_1_AES256.yml
01:14 Trying to use vault secret=(ScriptVaultSecret(filename='/root/ansible/test/results/.tmp/integration/ansible-vault-jbtvhg5w-ÅÑŚÌβŁÈ/test/integration/targets/ansible-vault/unicorn-password-script.py')) id=default to decrypt format_1_1_AES256.yml
01:14 Trying secret ScriptVaultSecret(filename='/root/ansible/test/results/.tmp/integration/ansible-vault-jbtvhg5w-ÅÑŚÌβŁÈ/test/integration/targets/ansible-vault/unicorn-password-script.py') for vault_id=default
01:14 Tried to use the vault secret (default) to decrypt (format_1_1_AES256.yml) but it failed. Error: HMAC verification failed: Signature did not match digest.
```
But upon closer inspection, I see (you are correct) that it did take the original route. I was thrown off by the `get_password_from_file()`.

Reverted all changes to `lib/ansible/cli/__init__.py` because the behavioural changes to `--connection-password-file` and `--become-password-file` are not mandatory for this change.

Reply:
> Tests should be added for --connection-password-file and --become-password-file in a separate PR before continuing with this one.

I'll leave those for now, since I'm not touching their behaviour.